### ISSUE TITLE
Version 3.20.0 changes JQL response format

### ISSUE THREAD

**Issue by predominant** [atlassian-api/atlassian-python-api#944]
Hi @gonchik - The v3.20.0 released 6 hours ago changes the format/type of the JQL() response on the jira API.

Before the release, it would return a dict, now it returns a list..

After change:
```
[DEBUG] HTTP: Response text -> {"startAt":0,"maxResults":50,"total":0,"issues":[]}
[DEBUG] Response Type <class 'list'>
```

Before change:
```
[DEBUG] HTTP: Response text -> {"startAt":0,"maxResults":50,"total":0,"issues":[]}
[DEBUG] Response Type <class 'dict'>
```

I think if this is to be the format moving forward, then this warrants a major version bump to 4.x, leaving the original JWL response format on the 3.x major version.

**Comment by poyntesm** [comment]
I agree. This has broken 3.x line. I am currently hardcoded jenkins builds to be pinned now to 3.19

**Comment by dsuch** [comment]
Hello @gonchik,

thanks for creating the library, it is a very helpful one indeed!

I am here to add my voice too - this commit 7f8bc5a4438fc6e555aca35d36735c53decfeddc significantly changed the output of the `.jql` method, a public one.

If that was not intentional, could you please revert it? If this is related to a use-case that you need to support, can you please make the new format optional or move it to another method, e.g. `jql_as_list`.

I assume that such core functionality as JQL is used by many projects that depend on your library.

Thank you.

**Comment by Spacetown** [comment]
Are you using the advanced mode of the API? I think the result should be a list or generator and the paging should be done internally. The advanced mode was needed to do the paging on client side.

**Comment by gonchik** [comment]
Hi! 
I do support the idea of @Spacetown. 
Also, made 2 method to the wrap, implemented in 3.20.1 release

**Comment by gonchik** [comment]
Hi @predominant @poyntesm ,
As I remember it has been fixed.
I am going to close that ticket.

if you want to improve that package feel free to send a PR
